src/mods/inputdisp.cpp

```diff
@@ -138,43 +112,39 @@ void init() {
                          [](f32 x, f32 y) { s_create_speed_sprites_tramp.dest(x + 5, y); });
 }
 
-void on_PADRead(mkb::PADStatus* statuses) {
-    mkb::memcpy(s_raw_inputs, statuses, sizeof(s_raw_inputs));
-}
-
 void tick() {
     s_rainbow = (s_rainbow + 3) % 1080;
     set_sprite_visible(!pref::get(pref::BoolPref::InputDisp) ||
                        (pref::get(pref::BoolPref::InputDispCenterLocation) &&
                         !pref::get(pref::BoolPref::InputDispRawStickInputs)));
 }
 
-static bool get_notch_pos(const MergedStickInputs& stick_inputs, Vec2d* out_pos) {
+static bool get_notch_pos(const pad::StickState& stick_inputs, Vec2d* out_pos) {
     constexpr f32 DIAG = 0.7071067811865476f;  // sin(pi/4) or sqrt(2)/2
     bool notch_found = false;
 
-    if (stick_inputs.gameX == 0 && stick_inputs.gameY == 60) {
+    if (stick_inputs.x == 0 && stick_inputs.y == 60) {
         *out_pos = {0, 1};
         notch_found = true;
-    } else if (stick_inputs.gameX == 0 && stick_inputs.gameY == -60) {
+    } else if (stick_inputs.x == 0 && stick_inputs.y == -60) {
         *out_pos = {0, -1};
         notch_found = true;
-    } else if (stick_inputs.gameX == 60 && stick_inputs.gameY == 0) {
+    } else if (stick_inputs.x == 60 && stick_inputs.y == 0) {
         *out_pos = {1, 0};
         notch_found = true;
-    } else if (stick_inputs.gameX == -60 && stick_inputs.gameY == 0) {
+    } else if (stick_inputs.x == -60 && stick_inputs.y == 0) {
         *out_pos = {-1, 0};
         notch_found = true;
-    } else if (stick_inputs.gameX == 60 && stick_inputs.gameY == 60) {
+    } else if (stick_inputs.x == 60 && stick_inputs.y == 60) {
         *out_pos = {DIAG, DIAG};
         notch_found = true;
-    } else if (stick_inputs.gameX == 60 && stick_inputs.gameY == -60) {
+    } else if (stick_inputs.x == 60 && stick_inputs.y == -60) {
         *out_pos = {DIAG, -DIAG};
         notch_found = true;
-    } else if (stick_inputs.gameX == -60 && stick_inputs.gameY == 60) {
+    } else if (stick_inputs.x == -60 && stick_inputs.y == 60) {
         *out_pos = {-DIAG, DIAG};
         notch_found = true;
-    } else if (stick_inputs.gameX == -60 && stick_inputs.gameY == -60) {
+    } else if (stick_inputs.x == -60 && stick_inputs.y == -60) {
         *out_pos = {-DIAG, -DIAG};
         notch_found = true;
     }
```

Comment:
These can use `MAX_STICK` too (might want to make those constants integer types and cast to float when needed)